### ISSUE TITLE
Follow up on COG S3 Like support

### ISSUE THREAD

**Issue by dromagnoli** [geosolutions-it/imageio-ext#223]
Improve S3 support to take into account S3 like storages (Minio, Cohesity).

**Comment by dnavre** [comment]
As I understand there has been no progress on the implementation of pathStyleAccess for s3? I need that feature in Geoserver but after struggling to make it work for several hours I ended up here :) Is there a roadmap of when this will be available? Or maybe some directions of how I should proceed if I'm going to implement this myself?

**Comment by ColorMaestro** [comment]
Is there any other syntax in which I could write the URL to make it work?

**Comment by dnavre** [comment]
> Is there any other syntax in which I could write the URL to make it work?

No, a change to the code-base has to be done. Currently there's no support for path style access, hence Minio, Ceph and others are also not supported.

**Comment by drndos** [comment]
Looks like it is possible like this:

- specify env variables:
```
      IIO_S3_AWS_USER: minio
      IIO_S3_AWS_PASSWORD: password
      IIO_S3_AWS_ENDPOINT: "http://192.168.1.120:9000"
      IIO_S3_AWS_REGION: "us-east-1"
```
- add geotiff repository with url: `cog://s3://bucket-name/cog_georeferenced.tif`

So far I have some issues with `Caused by: org.geotools.data.DataSourceException: Raster to Model Transformation is not available for: null` but that might be issue with my geotiff file.

**Comment by ColorMaestro** [comment]
@drndos
Thank you for the hit, creation of coverage store work now.

For your files, make sure they are in compatible format, see [https://www.cogeo.org/developers-guide.html](https://www.cogeo.org/developers-guide.html)

**Comment by vuvkar** [comment]
@drndos thank you for your update.
I have my env configured with the params as you mentioned above, but it's still trying to reach with virtual paths. I am using ceph.
Is there any additional configs that should have been done? :///
